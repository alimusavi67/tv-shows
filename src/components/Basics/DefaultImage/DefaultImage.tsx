import React from 'react';

interface DefaultImageProps {
  src: string | undefined;
  alt: string;
  className?: string;
}

const DefaultImage: React.FC<DefaultImageProps> = ({ src, alt, className = '' }) => (
  <img
    src={src || '/no-image.jpg'}
    alt={alt}
    data-testid="skeleton"
    className={`w-full rounded h-40 object-cover transition-transform duration-300 group-hover:scale-105 ${className}`}
  />
);

export default DefaultImage;
