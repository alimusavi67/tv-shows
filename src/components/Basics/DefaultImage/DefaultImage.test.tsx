import React from 'react';
import { render } from '@testing-library/react';
import DefaultImage from './DefaultImage';

describe('DefaultImage component', () => {
  test('renders correctly with src and alt', () => {
    const { asFragment } = render(<DefaultImage src="https://example.com/image.jpg" alt="Example Image" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders correctly with default image when src is undefined', () => {
    const { asFragment } = render(<DefaultImage src={undefined} alt="Example Image" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders correctly with additional className', () => {
    const { asFragment } = render(
      <DefaultImage src="https://example.com/image.jpg" alt="Example Image" className="custom-class" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
