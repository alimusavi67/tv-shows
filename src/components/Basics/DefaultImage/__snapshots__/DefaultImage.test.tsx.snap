// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultImage component renders correctly with additional className 1`] = `
<DocumentFragment>
  <img
    alt="Example Image"
    class="w-full rounded h-40 object-cover transition-transform duration-300 group-hover:scale-105 custom-class"
    data-testid="skeleton"
    src="https://example.com/image.jpg"
  />
</DocumentFragment>
`;

exports[`DefaultImage component renders correctly with default image when src is undefined 1`] = `
<DocumentFragment>
  <img
    alt="Example Image"
    class="w-full rounded h-40 object-cover transition-transform duration-300 group-hover:scale-105 "
    data-testid="skeleton"
    src="/no-image.jpg"
  />
</DocumentFragment>
`;

exports[`DefaultImage component renders correctly with src and alt 1`] = `
<DocumentFragment>
  <img
    alt="Example Image"
    class="w-full rounded h-40 object-cover transition-transform duration-300 group-hover:scale-105 "
    data-testid="skeleton"
    src="https://example.com/image.jpg"
  />
</DocumentFragment>
`;
